cloudwatch_logger/src/log_node.cpp

```diff
@@ -36,10 +36,10 @@ LogNode::LogNode(int8_t min_log_severity)
 LogNode::~LogNode() { this->log_manager_ = nullptr; }
 
 void LogNode::Initialize(const std::string & log_group, const std::string & log_stream,
-                         Aws::Client::ClientConfiguration & config, Aws::SDKOptions & sdk_options)
+                         const Aws::Client::ClientConfiguration & config, Aws::SDKOptions & sdk_options,
+                         const std::shared_ptr<LogManagerFactory> & factory)
 {
-  LogManagerFactory factory;
-  this->log_manager_ = factory.CreateLogManager(log_group, log_stream, config, sdk_options);
+  this->log_manager_ = factory->CreateLogManager(log_group, log_stream, config, sdk_options);
```

Comment:
When `Initialize()` is called without specifying `factory` (so it takes on the default temporary parameter), `factory` will just produce this one log manager and that's its entire lifetime. Is that the intention?

Reply:
Yes, this is replacing a local variable for the factory, by moving it to a parameter we can replace the production factory with a mock for the tests. The lifetime is the same as in the original local variable